kernel/src/process.rs

```diff
@@ -416,6 +420,11 @@ pub trait ProcessType {
             -> Result<ReadOnlyAppSlice, (ReadOnlyAppSlice, ErrorCode)>,
     ) -> GenericSyscallReturnValue;
 
+    /// Set a single byte within the process address space at
+    /// `addr` to `value`. Return true if `addr` is a valid
+    /// address and the value was written, false otherwise.
```

Comment:
In `process.rs` we have to be careful about what guarantees and preconditions the various methods enforce, to not risk writing to some other (potentially kernel-reserved) memory region. Maybe it's a good idea to copy parts of the documentation of `in_app_owned_memory` to also apply here?:

```suggestion
    /// Set a single byte within the process address space at
    /// `addr` to `value`. Return true if `addr` is within the RAM
    /// bounds currently exposed to the process (thereby writable
    /// by the process itself) and the value was set, false otherwise.
```